.github/workflows/workflow_finetune.yml

```diff
@@ -70,122 +70,46 @@ jobs:
-          cid=$(docker ps -a -q --filter "name=finetune")
-          if [[ ! -z "$cid" ]]; then docker rm $cid; fi
-          docker run -tid -v ${{ inputs.model_cache_path }}:/root/.cache/huggingface/hub -v ${{ inputs.code_checkout_path }}:/root/llm-on-ray -e http_proxy=${{ inputs.http_proxy }} -e https_proxy=${{ inputs.https_proxy }} --name="finetune" --hostname="finetune-container" finetune:latest
-
+          TARGET="finetune"
+          code_checkout_path=${{ inputs.code_checkout_path }}
+          model_cache_path=${{ inputs.model_cache_path }}
+          source dev/scripts/docker.sh
+          start_docker ${TARGET} ${code_checkout_path} ${model_cache_path} "USE_PROXY"
+          
       - name: Run Finetune Test
         run: |
-          docker exec "finetune" bash -c "source \$(python -c 'import oneccl_bindings_for_pytorch as torch_ccl;print(torch_ccl.cwd)')/env/setvars.sh; RAY_SERVE_ENABLE_EXPERIMENTAL_STREAMING=1 ray start --head --node-ip-address 127.0.0.1 --ray-debugger-external; RAY_SERVE_ENABLE_EXPERIMENTAL_STREAMING=1  ray start --address='127.0.0.1:6379' --ray-debugger-external"
-          CMD=$(cat << EOF
-          import yaml
-          conf_path = "llm_on_ray/finetune/finetune.yaml"
-          with open(conf_path, encoding="utf-8") as reader:
-              result = yaml.load(reader, Loader=yaml.FullLoader)
-              result['General']['base_model'] = "${{ matrix.model }}"
-              if "${{ matrix.model }}" == "mosaicml/mpt-7b":
-                  result['General']['config']['trust_remote_code'] = True
-              else:
-                  result['General']['config']['trust_remote_code'] = False
-              if "${{ matrix.model }}" == "EleutherAI/gpt-j-6b" or "${{ matrix.model }}" == "gpt2":
-                  result['General']["gpt_base_model"] = True
-              else:
-                  result['General']["gpt_base_model"] = False
-              if "${{ matrix.model }}" == "meta-llama/Llama-2-7b-chat-hf" or "${{ matrix.model }}" == "google/gemma-2b":
-                  result['General']["config"]["use_auth_token"] = "${{ env.HF_ACCESS_TOKEN }}"
-              else:
-                  result['General']["config"]["use_auth_token"] = None
-              result['Training']['epochs'] = 1
-              if "${{ matrix.model }}" == "gpt2":
-                  # to verify oneccl
-                  result['Training']['num_training_workers'] = 2
-              else:
-                  result['Training']['num_training_workers'] = 1
-              result['General']['lora_config'] = None
-          with open(conf_path, 'w') as output:
-              yaml.dump(result, output, sort_keys=False)
-          EOF
-          )
-          docker exec "finetune" python -c "$CMD"
-          docker exec "finetune" bash -c "llm_on_ray-finetune --config_file llm_on_ray/finetune/finetune.yaml"
+          TARGET="finetune"
+          docker exec "${TARGET}" bash -c "huggingface-cli login --token ${{ env.HF_ACCESS_TOKEN }}"
```

Comment:
could we move huggingface login to docker container start as it will be used for all the steps of test.